engine/common/synchronization/engine.go

```diff
@@ -321,50 +293,45 @@ func (e *Engine) onBlockResponse(originID flow.Identifier, res *messages.BlockRe
 }
 
 // checkLoop will regularly scan for items that need requesting.
-func (e *Engine) checkLoop() {
+func (e *Engine) checkLoop(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
 	pollChan := make(<-chan time.Time)
 	if e.pollInterval > 0 {
 		poll := time.NewTicker(e.pollInterval)
 		pollChan = poll.C
 		defer poll.Stop()
 	}
 	scan := time.NewTicker(e.scanInterval)
+	defer scan.Stop()
 
+	done := ctx.Done()
 CheckLoop:
```

Comment:
since we don't need to do any additional cleanup, it would be best to remove the label and just return from function instead of using `break` statement